### ISSUE TITLE
Low priority write transaction

### ISSUE THREAD

**Issue by haikalpribadi** [graknlabs/grakn#6157]
## Problem to Solve

When migrating lots of data into a live running server, it's possible to hog all the computation resource, as well as causing write-stalls from RocksDB, if we're ingesting data faster than RocksDB can do compaction for in time. This would cause the normal user-queries to stall too, which would be undesirable.

## Current Workaround

None.

## Proposed Solution

Introduce a "low priority" write transaction, such that when data loads coming from these transactions cause to overflow the machine, only these write transactions will get throttled, and not the rest (user-queries).

1) Introduce a "secondary thread pool" which is smaller than the main thread pool.
2) Use set RocksDB transaction to "low-priority" for all transactions in this category.

## Additional Information

https://github.com/facebook/rocksdb/wiki/Low-Priority-Write

**Comment by chris-aeviator** [comment]
How can I archive this? The docs do not mention any strategies howto mass ingest data - since typedb is intended to be usable with machine learning this is a dealbreaker for me.

**Comment by haikalpribadi** [comment]
Why would we want to archive this issue @chris-aeviator ? It exists because we want to implement it at some point. This issue is also an optimisation. It shouldn't be a blocker at this point.

**Comment by chris-aeviator** [comment]
With some research and chat in discord I found a way to mass ingest. It seems not well documented here what roles the single parts of a) creating a session b) creating a transaction c) writing d) closing of each

Play. I ended up choosing a specific pattern for looping over my data and ingested 3000 nodes per transaction (total data size 75k entries) which was an (un-)educated guess. The docs display a very simple usecase/ example only and are not meant to show production usage. I think adding more/ more realistic client examples would greatly benefit newcomers to get started and not abandon this project half way through their dev process.

**Comment by haikalpribadi** [comment]
Yes, completely see what you mean, @chris-aeviator. We're just about to rewrite the whole documentation to be much more cohesive and informative, and it will be done very soon! Hang in there!
